db/table_cache.cc

```diff
@@ -100,6 +100,10 @@ Status TableCache::GetTableReader(
   std::unique_ptr<FSRandomAccessFile> file;
   Status s = ioptions_.fs->NewRandomAccessFile(fname, file_options, &file,
                                                nullptr);
+  if (s.IsPathNotFound()) {
```

Comment:
One minor thing, you probably need to call `RecordTick(ioptions_.statistics, NO_FILE_OPENS)` even for failed file open attempts.